packages/world/src/modules/batchcall/BatchCallSystem.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import { System } from "../../System.sol";
+import { IBaseWorld } from "../../interfaces/IBaseWorld.sol";
+
+contract BatchCallSystem is System {
+  function batchCall(bytes32[] memory resourceSelectors, bytes[] memory funcSelectorAndArgss) public {
+    IBaseWorld world = IBaseWorld(_world());
+
+    for (uint256 i; i < resourceSelectors.length; i++) {
+      world.callFrom(_msgSender(), resourceSelectors[i], funcSelectorAndArgss[i]);
```

Comment:
should we make this a root system and then do this so the users don't need to create a delegation?

```solidity
(bool success, bytes memory returnData) = address(world).delegatecall(abi.encodeCall(
  world.call,
  (
    _msgSender(),
    resourceSelector[i],
    funcSelectorAndArgs[i]
  )
));
if(!success) revertWithBytes(returnData);
```

We can actually do both, have a `BatchCallSystem` and a `BatchCallRootSystem`, and install then in `installModule` and `installRootModule` respectively

Reply:
do we need this to be a module? if we intend to use this for deploys anyway, would it make sense to just have this under `core/implementations` as a plain root system?

Reply:
yeah it can also be part of the core module

Reply:
I like this approach without the delegation. I'm going to add the root version to the core module. 

I'll also remove the non-root version, because the root version will be installed on worlds by default.